### PR TITLE
Fix #1296 Add "Sign in with Slack (OpenID Connect)" support

### LINKED ISSUE #1296: Add "Sign in with Slack (OpenID Connect)" support
### Description

The new version of Sign in with Slack has been released! On the SDK side, we can do the following things for better support of the functionality:

* Add Web API supports
  * openid.connect.token
  * openid.connect.userInfo
* Share a fully functioning example app

python-slack-sdk pr adding support for reference is at https://github.com/slackapi/python-slack-sdk/pull/1079

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [ ] bug
- [x] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

### PR DESCRIPTION
**Pull request by seratch**
###  Summary

This pull request fixes #1296 by adding the following updates:

* Add `openid.connect.*` endpoint supports with generated API response types
* Add an example app demonstrating how to implement Slack OpenID Connect

I will add a few comments for reviewers.

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: examples/openid-connect/app.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+const Koa = require("koa");
```
Comment (seratch) [inline examples/openid-connect/app.js:1]: Currently, we have only Koa app example. I don't think we should have both (because there is no significant difference) but would like to know others' thoughts.

File: examples/openid-connect/link.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (seratch) [inline examples/openid-connect/link.sh:1]: You can run this script to easily run the example. Once we release v6.4, this will be no longer needed.

File: packages/web-api/src/methods.ts [hunk 3]
```diff
@@ -188,6 +188,8 @@ import { AdminUsersSessionSetSettingsResponse } from './response/AdminUsersSessi
 import { AdminUsersSessionClearSettingsResponse } from './response/AdminUsersSessionClearSettingsResponse';
 import { AdminAppsClearResolutionResponse } from './response/AdminAppsClearResolutionResponse';
 import { AdminAppsUninstallResponse } from './response/AdminAppsUninstallResponse';
+import { OpenidConnectTokenResponse } from './response/OpenidConnectTokenResponse';
```
Comment (seratch) [inline packages/web-api/src/methods.ts:191]: If we want to adjust the name to `OpenIDConnect`, you can do it by modifying the `scripts/code_generator.rb`

File: packages/web-api/src/response/ConversationsHistoryResponse.ts [hunk 4]
```diff
@@ -82,6 +82,7 @@ export interface Attachment {
   thumb_url?:             string;
   thumb_width?:           number;
   thumb_height?:          number;
+  video_url?:             string;
```
Comment (seratch) [inline packages/web-api/src/response/ConversationsHistoryResponse.ts:85]: These unrelated changes are the ones detected by recent. integration tests in the Java SDK project

**Comment by seratch** [comment]
Any comments?

File: examples/openid-connect/README.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+```yaml
+_metadata:
+  major_version: 1
+  minor_version: 1
+display_information:
+  name: my-openid-connect-app
+oauth_config:
+  redirect_urls:
+    # TODO: Replace the URL with our own one
```
Comment (filmaj) [inline examples/openid-connect/README.md:17]:
```suggestion
    # TODO: Replace the URL with your own
```

File: examples/openid-connect/README.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+## Install Dependencies
+
+```
+npm install
+```
+
+## Setup Environment Variables
+
+This app requires you setup a few environment variables. You can get these values by navigating to your app's [**BASIC INFORMATION** Page](https://api.slack.com/apps).
```
Comment (filmaj) [inline examples/openid-connect/README.md:34]:
```suggestion
This app requires you to setup a few environment variables. You can get these values by navigating to your app's [**BASIC INFORMATION** Page](https://api.slack.com/apps).
```

File: examples/openid-connect/app.js (hunk 1 above)
Comment (filmaj) [reply examples/openid-connect/app.js:1]: When you say "both", what do you mean?

File: examples/openid-connect/app.js [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+      )}`
+    );
+    userAccessToken = refreshedToken.access_token;
+  }
+
+  // You can save the id_token (JWT string) as-is but you can decode the value this way:
+  const claims = jwt.decode(token.id_token);
+
+  // TODO: you can do something meaningful here
+  // (e.g., storing the data in database + navigating the user to your service top paeg)
```
Comment (filmaj) [inline examples/openid-connect/app.js:108]:
```suggestion
  // (e.g., storing the data in database + navigating the user to your service top page)
```

File: examples/openid-connect/app_manifest.yml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+_metadata:
+  major_version: 1
+  minor_version: 1
+display_information:
+  name: my-openid-connect-app
+oauth_config:
+  redirect_urls:
+    # TODO: Replace the URL with our own one
```
Comment (filmaj) [inline examples/openid-connect/app_manifest.yml:8]:
```suggestion
    # TODO: Replace the URL with your own one
```

File: packages/web-api/src/methods.ts (hunk 3 above)
Comment (filmaj) [reply packages/web-api/src/methods.ts:191]: I personally prefer `OpenID` or `OpenId` but that is just a preference.

**Review by filmaj — APPROVED** [review]
Looks great, the example application is super helpful. I left a few small suggestions.

File: examples/openid-connect/README.md (hunk 6 above)
Comment (seratch) [reply examples/openid-connect/README.md]: Thanks. This sentence is taken from https://github.com/slackapi/node-slack-sdk/blob/main/examples/oauth-v2/README.md - we may want to update the original one too

File: examples/openid-connect/app.js (hunk 1 above)
Comment (seratch) [reply examples/openid-connect/app.js:1]: I mean having two examples - koa and express.

**Review by seratch — COMMENTED** [review]
Thanks for the review!

File: scripts/code_generator.rb [hunk 9]
```diff
@@ -54,6 +54,10 @@ def append_to_index(root_class_name, index_file)
       end
       prev_c = c
     end
+    if root_class_name.start_with? 'Openid'
+      root_class_name.sub!('Openid', 'OpenID')
```
Comment (seratch) [inline scripts/code_generator.rb:58]: @filmaj I changed my mind. The generator script replaces `Openid` with `OpenID` now.

File: examples/openid-connect/README.md (hunk 6 above)
Comment (filmaj) [reply examples/openid-connect/README.md:34]: Sounds good, I fixed it in https://github.com/slackapi/node-slack-sdk/commit/74149a50efa0d0219670115ef1a2e3ac7ccbcd04

File: examples/openid-connect/app.js (hunk 1 above)
Comment (filmaj) [reply examples/openid-connect/app.js:1]: Cool, thanks for the clarification. I will resolve this comment.

File: scripts/code_generator.rb (hunk 9 above)
Comment (filmaj) [reply scripts/code_generator.rb:58]: Nice!

**Comment by seratch** [comment]
@filmaj Thanks for your prompt review! Let me merge this one now.
